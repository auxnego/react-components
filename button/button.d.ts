import { PropsWithChildren, AllHTMLAttributes } from "react";

//
import {
  BasicConfig,
  ClassNameConfig
} from "../global.d";

/** Button Component **/

export type ButtonMainProps = {};

export type ButtonProps = PropsWithChildren &
                          BasicConfig &
                          ClassNameConfig &
                          AllHTMLAttributes<HTMLElement>
                          ButtonMainProps;
