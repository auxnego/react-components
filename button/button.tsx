import React, { Attributes, MouseEventHandler } from "react";
import clsx from "clsx";

//types
import {
  ButtonProps
} from "./button.d";

export default function Button(props:ButtonProps){
  //default props
  let {
    children = null,
    className = "",
    classless = false,
    setEl = "button",
    role = "button",
    disabled = false,
    onClick,
    ...more
  } = props;

  let __handleClick:MouseEventHandler<HTMLElement> = (e) => {
    //actions are prohibts when button is disabled
    if(!disabled){
      if(onClick) onClick(e);
    }
  };

  //classNames
  let __className = [!classless && {
    "button":true,
    "button--disabled":!!disabled
  }, className];

  //props
  let __props = {
    ...more,
    role,
    className:clsx(__className),
    disabled,
    onClick:__handleClick
  } as Attributes;

  //views
  return React.createElement(setEl, __props, children);
}
