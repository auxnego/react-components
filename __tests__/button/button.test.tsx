import React from "react";
import {create} from "react-test-renderer";

import { Button } from "../../button";

it("simple button", () => {

  expect(create(
    <Button>Aceptar</Button>
  ).toJSON()).toMatchSnapshot();

});

it("disabled button", () => {

  expect(create(
    <Button disabled>Disabled Button</Button>
  ).toJSON()).toMatchSnapshot();

});

it("setEl=div button", () => {

  expect(create(
    <Button setEl="div">Aceptar</Button>
  ).toJSON()).toMatchSnapshot();

});

