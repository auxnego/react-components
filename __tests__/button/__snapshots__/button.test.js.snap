// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`disabled button 1`] = `
<button
  className="button button--disabled"
  disabled={true}
  onClick={[Function]}
  role="button"
>
  Disabled Button
</button>
`;

exports[`setEl=div button 1`] = `
<div
  className="button"
  disabled={false}
  onClick={[Function]}
  role="button"
>
  Aceptar
</div>
`;

exports[`simple button 1`] = `
<button
  className="button"
  disabled={false}
  onClick={[Function]}
  role="button"
>
  Aceptar
</button>
`;
